spec/services/db/monthly_sp_auth_count/unique_monthly_auth_counts_by_partner_spec.rb

```diff
@@ -0,0 +1,247 @@
+require 'rails_helper'
+
+RSpec.describe Db::MonthlySpAuthCount::UniqueMonthlyAuthCountsByPartner do
+  describe '.call' do
+    subject(:results) do
+      Db::MonthlySpAuthCount::UniqueMonthlyAuthCountsByPartner.call(
+        key: key,
+        start_date: partner_range.begin,
+        end_date: partner_range.end,
+        issuers: issuers,
+      )
+    end
+
+    context 'with no data' do
+      let(:key) { partner }
+      let(:partner) { 'DHS' }
+      let(:partner_range) { 1.year.ago..Time.zone.now }
+      let(:issuers) { [] }
+
+      it 'is empty with no data issuers' do
+        expect(results).to eq([])
+      end
+    end
+
+    context 'with data' do
+      let(:key) { 'DHS' }
+      let(:issuers) { [issuer1, issuer2, issuer3] }
+
+      let(:partner_range) { Date.new(2020, 9, 15)..Date.new(2021, 9, 14) }
+      let(:inside_partial_month) { Date.new(2020, 9, 16) }
+      let(:inside_whole_month) { Date.new(2020, 10, 16) }
+
+      let(:user1) { create(:user, profiles: [profile1a, profile1b]) }
+      let(:profile1a) { create(:profile, verified_at: '2015-09-16') }
+      let(:profile1b) { create(:profile, verified_at: '2020-09-16') }
+
+      let(:user2) { create(:user, profiles: [profile2]) }
+      let(:profile2) { create(:profile, verified_at: '2018-12-25') }
+
+      let(:user3) { create(:user, profiles: [profile3]) }
+      let(:profile3) { create(:profile, verified_at: '2019-11-10') }
+
+      let(:user4) { create(:user, profiles: [profile4]) }
+      let(:profile4) { create(:profile, verified_at: '2020-03-01') }
+
+      let(:user5) { create(:user, profiles: [profile5]) }
+      let(:profile5) { create(:profile, verified_at: '2019-04-17') }
+
+      let(:user6) { create(:user, profiles: [profile6]) }
+      let(:profile6) { create(:profile, verified_at: '2018-09-15') }
+
+      let(:user7) { create(:user, profiles: [profile7]) }
+      let(:profile7) { create(:profile, verified_at: '2017-02-15') }
+
+      let(:user8) { create(:user, profiles: [profile8]) }
+      let(:profile8) { create(:profile, verified_at: '2016-03-20') }
+
+      let(:user9) { create(:user, profiles: [profile9]) }
+      let(:profile9) { create(:profile, verified_at: '2012-12-15') }
+
+      let(:user10) { create(:user, profiles: [profile10]) }
+      let(:profile10) { create(:profile, verified_at: nil) }
+
+      let(:user11) { create(:user, profiles: [profile9]) }
+      let(:profile11) { create(:profile, verified_at: '2019-11-10') }
```

Comment:
Let's switch all the `create(:profile)` to `build(:profile)`

The difference is that `create(:profile)` will also create its own user object in the database, so doing `build(:profile)` will create a user object in memory but won't write it to the database

```suggestion
      let(:profile1a) { build(:profile, verified_at: '2015-09-16') }
      let(:profile1b) { build(:profile, verified_at: '2020-09-16') }

      let(:user2) { create(:user, profiles: [profile2]) }
      let(:profile2) { build(:profile, verified_at: '2018-12-25') }

      let(:user3) { create(:user, profiles: [profile3]) }
      let(:profile3) { build(:profile, verified_at: '2019-11-10') }

      let(:user4) { create(:user, profiles: [profile4]) }
      let(:profile4) { build(:profile, verified_at: '2020-03-01') }

      let(:user5) { create(:user, profiles: [profile5]) }
      let(:profile5) { build(:profile, verified_at: '2019-04-17') }

      let(:user6) { create(:user, profiles: [profile6]) }
      let(:profile6) { build(:profile, verified_at: '2018-09-15') }

      let(:user7) { create(:user, profiles: [profile7]) }
      let(:profile7) { build(:profile, verified_at: '2017-02-15') }

      let(:user8) { create(:user, profiles: [profile8]) }
      let(:profile8) { build(:profile, verified_at: '2016-03-20') }

      let(:user9) { create(:user, profiles: [profile9]) }
      let(:profile9) { build(:profile, verified_at: '2012-12-15') }

      let(:user10) { create(:user, profiles: [profile10]) }
      let(:profile10) { build(:profile, verified_at: nil) }

      let(:user11) { create(:user, profiles: [profile9]) }
      let(:profile11) { build(:profile, verified_at: '2019-11-10') }
```

Reply:
Done!